eo-runtime/src/main/java/org/eolang/ExprReduce.java

```diff
@@ -74,71 +67,138 @@
      * Ctor.
      *
      * @param oper Operation name
-     * @param param Name of parameter with varargs
-     * @param type Type of varargs
-     * @param reduction Reduction operation on consecutive varags
-     * @param validation Validation function
+     * @param param Param name with varargs
+     * @param arguments Object contains type of varargs, validation and reduction functions
      * @checkstyle ParameterNumberCheck (10 lines)
      */
     public ExprReduce(
         final String oper,
         final String param,
-        final Class<T> type,
-        final BiFunction<T, T, T> reduction,
-        final Function<T, String> validation
+        final Args<T> arguments
     ) {
-        this.param = param;
-        this.type = type;
-        this.reduction = reduction;
         this.oper = oper;
-        this.validation = validation;
-    }
-
-    /**
-     * Ctor.
-     * @param oper Peration name
-     * @param param Name of parameter with varargs
-     * @param type Type of varargs
-     * @param reduction Reduction operation on consecutive varargs
-     * @checkstyle ParameterNumberCheck (10 lines)
-     */
-    public ExprReduce(
-        final String oper,
-        final String param,
-        final Class<T> type,
-        final BiFunction<T, T, T> reduction
-    ) {
-        this(
-            oper,
-            param,
-            type,
-            reduction,
-            x -> ""
-        );
+        this.param = param;
+        this.arguments = arguments;
     }
-
     @Override
     public Phi get(final Phi rho) {
-        T acc = new Param(rho).strong(this.type);
+        T acc = this.arguments.accumulator(rho);
```

Comment:
@potatmen this can be eliminated at all (see comments below)

Reply:
@mximp ok, I will change it